javascript/extractor/src/com/semmle/js/parser/TypeScriptParser.java

```diff
@@ -236,9 +248,20 @@ private void setupParserWrapper() {
     int reserveMemoryMb = getMegabyteCountFromPrefixedEnv(TYPESCRIPT_RAM_RESERVE_SUFFIX, 400);
 
     File parserWrapper = getParserWrapper();
-    
+
+    String debuggerFlag = Env.systemEnv().get(TYPESCRIPT_ATTACH_DEBUGGER);
+    String inspectArg = "";
+    if (debuggerFlag != null && debuggerFlag.length() > 0) {
```

Comment:
Why not map the arguments tighter to the arguments that Node understands? This will make it easier to generalise later. You already describe how they are mapped above.

The security-paranoid version is this:

```suggestion
      switch (debuggerFlag) {
        case "--inspect-brk": {
          inspectArg = "--inspect-brk";
          break;
        }
        case "--inspect": {
          inspectArg = "--inspect";
          break;
        }
        // ... 
      }
```